data/helpers.d/apt

```diff
@@ -218,6 +218,27 @@ ynh_install_app_dependencies () {
     fi
     local dep_app=${app//_/-}	# Replace all '_' by '-'
 
+    #
+    # Epic ugly hack to fix the goddamn dependency nightmare of sury
+    # Sponsored by the "Djeezusse Fokin Kraiste Why Do Adminsys Has To Be So Fucking Complicated I Should Go Grow Potatoes Instead Of This Shit" collective
+    # https://github.com/YunoHost/issues/issues/1407
+    # 
+    # If we require to install php dependency
+    if echo $dependencies | grep -q 'php';
+    then
+        # And we have packages from sury installed (7.0.33-10+weirdshiftafter instead of 7.0.33+1 on debian)
+        if dpkg --list | grep php | grep -q "7.0.33-10"
+        then
+            # And sury ain't already installed
+            if ! grep -nrq "sury" /etc/apt/sources.list*
```

Comment:
Ah, we need to look inside `/etc/apt/sources.list.d/`, no?

Reply:
Hmmmmyup but it should be caught in the `/etc/apt/sources.list*` (note the wildcard) ?

Reply:
Ah, I'm re-installing both apps again and I'll test this by hand once more. Two secs.

Reply:
```
root@ynhdev:~# dpkg --list | grep php | grep 7.0.33
ii  php7.0-cli                            7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        command-line interpreter for the PHP scripting language
ii  php7.0-common                         7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        documentation, examples and common module for PHP
ii  php7.0-curl                           7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        CURL module for PHP
ii  php7.0-fpm                            7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        server-side, HTML-embedded scripting language (FPM-CGI binary)
ii  php7.0-gd                             7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        GD module for PHP
ii  php7.0-intl                           7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        Internationalisation module for PHP
ii  php7.0-json                           7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        JSON module for PHP
ii  php7.0-ldap                           7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        LDAP module for PHP
ii  php7.0-mbstring                       7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        MBSTRING module for PHP
ii  php7.0-mcrypt                         7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        libmcrypt module for PHP
ii  php7.0-mysql                          7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        MySQL module for PHP
ii  php7.0-opcache                        7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        Zend OpCache module for PHP
ii  php7.0-readline                       7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        readline module for PHP
ii  php7.0-xml                            7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        DOM, SimpleXML, WDDX, XML, and XSL module for PHP
ii  php7.0-zip                            7.0.33-11+0~20190923.20+debian9~1.gbpd05c7e       amd64        Zip module for PHP
```

Where "-10" doesn't make the correct match ... :sweat:

Reply:
Hmwelll, I guess we need to either use a real version comparison tool to be more robust, or grep somethind else like the "+debian9~" part of the "gbp" thing (the hexadecimal stuff seems to be changing over time) or idk :/

Reply:
Oooooor `grep -v "7.0.33-0+deb9u5"` which is debian's version

Reply:
`dpkg --list | grep php | grep -v "7.0.33-0+deb9u5` does return what we need!

Reply:
Yes, I've patched that and I can isntall Nextcloud successfully. It should be fine :hand: